### ISSUE TITLE
Correctly Preparing Train Data for Tiny YOLOv2

### ISSUE THREAD

**Issue by formulajuan** [sassoftware/python-dlpy#251]
Hi,

I'm trying to train a single class object detector using Tiny YOLOv2 in DLPy. I have high-res jpg images and txt label files which I've uploaded to separate CAS Tables and then joined to create a 'trainSet' table [as outlined in the documentation on the SAS website](https://documentation.sas.com/?docsetId=casdlpg&docsetTarget=p1np8zbnoyd0brn1dhehthuuxj4q.htm&docsetVersion=8.3&locale=en#n19lxyzu2hmn1hn1v674oh55lf6u). My 'trainSet' table has the following column info:

  | Column | ID | Type | RawLength | FormattedLength | NFL | NFD
-- | -- | -- | -- | -- | -- | -- | --
0 | _filename_0 | 1 | varchar | 13 | 13 | 0 | 0
1 | _image_ | 2 | varbinary | 5528485 | 5528485 | 0 | 0
2 | _label_ | 3 | varchar | 0 | 1 | 0 | 0
3 | _id_ | 4 | int64 | 8 | 12 | 0 | 0
4 | _nObjects_ | 5 | double | 8 | 12 | 0 | 0
5 | _Object0_ | 6 | varchar | 7 | 7 | 0 | 0
6 | _Object0_x | 7 | double | 8 | 12 | 0 | 0
7 | _Object0_y | 8 | double | 8 | 12 | 0 | 0
8 | _Object0_width | 9 | double | 8 | 12 | 0 | 0
9 | _Object0_height | 10 | double | 8 | 12 | 0 | 0

Running through the rest of the GitHub example building the model and then using `model.set_weights` and `model.fit` seems to work until it reaches the point of processing the image data:

```
solver = MomentumSolver(learning_rate=0.001, clip_grad_max = 100, clip_grad_min = -100)
optimizer = Optimizer(algorithm=solver, mini_batch_size=64, log_level=2, max_epochs=10, reg_l2=0.005)
data_specs = [DataSpec(type_='IMAGE', layer='Input1', data=inputVars),
              DataSpec(type_='OBJECTDETECTION', layer='Detection1', data=targets)]
gpu = Gpu(devices=[0])
yolo_model.set_weights('TINY-YOLOV2_WEIGHTS')
yolo_model.fit(data='trainSet', 
               optimizer=optimizer, 
               data_specs=data_specs, 
               n_threads=1, 
               record_seed=13309,
               force_equal_padding=True 
#                gpu=gpu
              )
```
NOTE: Model weights attached successfully!
NOTE: Training based on existing weights.
WARNING: Using dataSpecs settings. Additional input or target option settings will be ignored.
NOTE:  Synchronous mode is enabled.
NOTE:  The total number of parameters is 11029392.
NOTE:  The approximate memory cost is 7098.00 MB.
NOTE:  Loading weights cost       0.56 (s).
NOTE:  Initializing each layer cost       2.12 (s).
NOTE:  The total number of threads on each worker is 1.
NOTE:  The total mini-batch size per thread on each worker is 64.
NOTE:  The maximum mini-batch size across all workers for the synchronous mode is 64.
NOTE:  Number of levels for the target variable:      1
NOTE:  Levels for the target variable:
NOTE:  Level      0: counter
**ERROR: The input image is the wrong type, or is missing the required width, height, or channel information.**
ERROR: The action stopped due to errors.

So I think the problem comes from loading the raw images to an ImageTable, but I'm not sure  what exactly is wrong with my ImageTable before using the dlJoin action to combine the images/labels into 'trainSet'.

The column info for my ImageTable is:

  | Column | ID | Type | RawLength | FormattedLength | NFL | NFD
-- | -- | -- | -- | -- | -- | -- | --
0 | _image_ | 1 | varbinary | 5528485 | 5528485 | 0 | 0
1 | _label_ | 2 | varchar | 0 | 1 | 0 | 0
2 | _filename_0 | 3 | varchar | 13 | 13 | 0 | 0
3 | _id_ | 4 | int64 | 8 | 12 | 0 | 0

This table was then joined with the CAS table containing the image labels by the id '_filename_0'.

**Comment by ethem-kinginthenorth** [comment]
what is your "inputVars" ? Also  yolo's default size is 416x416. Your input images should be 416x416 as well. They don't have to be 416x416 but they have to match.

**Comment by ethem-kinginthenorth** [comment]
there is also an example notebook to prepare data for object detection. We have a couple of APIs that can be used for this purpose: https://github.com/sassoftware/python-dlpy/tree/master/examples/object_detection

**Comment by formulajuan** [comment]
> what is your "inputVars" ? Also yolo's default size is 416x416. Your input images should be 416x416 as well. They don't have to be 416x416 but they have to match.

My 'inputVars' is ['_image_'].

I have manually converted the labels to YOLO format as they are provided in the format `[xmin, ymin, w, h]`. These labels correspond to the normal resolution photo, so if the training images have to be resized to 416x416 then the aspect ratio of the image changes and so do the labels, is that how the labels should be calculated in the first place?

**Comment by ethem-kinginthenorth** [comment]
Yolo format is normalized by width and height. Therefore, when you resize the images, bounding boxes still work. Thanks.

**Comment by formulajuan** [comment]
Okay I will try that, thank you!

**Comment by formulajuan** [comment]
Hi, this issue can be closed. Thanks for your help!
